examples/AdvancedOperations/AddAppCampaign.php

```diff
@@ -0,0 +1,443 @@
+     */
+    private static function setCampaignTargetingCriteria(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignResourceName
+    ) {
+        $campaignCriterionOperations = [];
+
+        // Creates the location campaign criteria.
+        // Besides using location_id, you can also search by location names from
+        // GeoTargetConstantService.suggestGeoTargetConstants() and directly
+        // apply GeoTargetConstant.resource_name here. An example can be found
+        // in GetGeoTargetConstantByNames.php.
+        $locationIds = [
+            21137, // California
+            2484   // Mexico
+        ];
+        foreach ($locationIds as $locationId) {
+            // Creates a campaign criterion.
+            $campaignCriterion = new CampaignCriterion([
+                'campaign' => new StringValue(['value' => $campaignResourceName]),
+                'type' => CriterionType::LOCATION,
+                'location' => new LocationInfo([
+                    'geo_target_constant' => new StringValue([
+                        'value' => ResourceNames::forGeoTargetConstant($locationId)
+                    ])
+                ])
+            ]);
+
+            // Creates a campaign criterion operation.
+            $campaignCriterionOperation = new CampaignCriterionOperation();
+            $campaignCriterionOperation->setCreate($campaignCriterion);
+
+            $campaignCriterionOperations[] = $campaignCriterionOperation;
+        }
+
+        // Creates the language campaign criteria.
+        $languageIds = [
+            1000, // English
+            1003  // Spanish
+        ];
+        foreach ($languageIds as $languageId) {
+            // Creates a campaign criterion.
+            $campaignCriterion = new CampaignCriterion([
+                'campaign' => new StringValue(['value' => $campaignResourceName]),
+                'type' => CriterionType::LANGUAGE,
+                'language' => new LanguageInfo([
+                    'language_constant' => new StringValue(['value' =>
+                        ResourceNames::forLanguageConstant($languageId)
+                    ])
+                ])
+            ]);
+
+            // Creates a campaign criterion operation.
+            $campaignCriterionOperation = new CampaignCriterionOperation();
+            $campaignCriterionOperation->setCreate($campaignCriterion);
+
+            $campaignCriterionOperations[] = $campaignCriterionOperation;
+        }
+
+        // Submits the criteria operations.
+        $campaignCriterionServiceClient = $googleAdsClient->getCampaignCriterionServiceClient();
+        $response = $campaignCriterionServiceClient->mutateCampaignCriteria(
+            $customerId,
+            $campaignCriterionOperations
+        );
+
+        $createdCampaignCriteriaResourceName = $response->getResults()[0]->getResourceName();
```

Comment:
This is probably wrong. We created many criteria. Why do we print only one?
I think Fan and David also fixed this. Could you see the final version of the file?

https://github.com/googleads/google-ads-python/pull/220/files

Reply:
The current version of the golden [does it like this](https://github.com/googleads/google-ads-python/blob/master/examples/advanced_operations/add_app_campaign.py#L227) so I changed for a listing instead.